pkg/reconciler/openshift/tektonconfig/extension.go

```diff
@@ -67,10 +68,37 @@ func (oe openshiftExtension) PreReconcile(ctx context.Context, tc v1alpha1.Tekto
 	r := rbac{
 		kubeClientSet:     oe.kubeClientSet,
 		operatorClientSet: oe.operatorClientSet,
-		ownerRef:          configOwnerRef(tc),
 		version:           os.Getenv(versionKey),
+		tektonConfig:      config,
 	}
-	return r.createResources(ctx)
+
+	var createResourceError error
+	if len(config.Spec.Params) == 0 {
+		createResourceError = r.createResources(ctx)
+	}
+	for _, v := range config.Spec.Params {
+		// check for param name and if its matches to createRbacResource
+		// then disable auto creation of RBAC resources by deleting installerSet
+		if v.Name == "createRbacResource" {
+			if v.Value == "false" {
+				if err := r.deleteInstallerSet(ctx, r.tektonConfig, componentName); err != nil {
+					return err
+				}
+				// remove openshift-pipelines.tekton.dev/namespace-reconcile-version label from namespaces while deleting RBAC resources.
+				if err := r.cleanUp(ctx); err != nil {
+					return err
+				}
+
+			} else {
+				createResourceError = r.createResources(ctx)
```

Comment:
we could directly return the error and return a `nil` at the end after the for loop